src/data/polynomial/degree/lemmas.lean

```diff
@@ -139,6 +139,40 @@ lemma nat_degree_lt_coeff_mul (h : p.nat_degree + q.nat_degree < m + n) :
   (p * q).coeff (m + n) = 0 :=
 coeff_eq_zero_of_nat_degree_lt (nat_degree_mul_le.trans_lt h)
 
+lemma coeff_mul_of_nat_degree_le (pm : p.nat_degree ≤ m) (qn : q.nat_degree ≤ n) :
+  (p * q).coeff (m + n) = p.coeff m * q.coeff n :=
+begin
+  refine (coeff_mul _ _ _).trans _,
+  refine finset.sum_eq_single (m, n) _ (by simp),
+  rintros ⟨d1, e1⟩ h de,
+  rcases (trichotomous d1 m : d1 < m ∨ _) with k|rfl|k,
+  { convert mul_zero _,
+    refine coeff_eq_zero_of_nat_degree_lt (qn.trans_lt _),
+    linarith [finset.nat.mem_antidiagonal.mp h] },
+  { exact (de (by simpa using h)).elim },
+  { convert zero_mul _,
+    exact coeff_eq_zero_of_nat_degree_lt (pm.trans_lt k) }
+end
+
+lemma coeff_pow_of_nat_degree_le (pn : p.nat_degree ≤ n) :
+  (p ^ m).coeff (n * m) = (p.coeff n) ^ m :=
+begin
+  induction m with m hm,
+  { simp },
+  { rw [pow_succ', pow_succ', ← hm, nat.mul_succ, coeff_mul_of_nat_degree_le _ pn],
+    refine nat_degree_pow_le.trans (le_trans _ (mul_comm _ _).le),
+    exact mul_le_mul_of_nonneg_left pn m.zero_le }
+end
+
+lemma coeff_add_eq_left_of_succ (n : ℕ) (qn : q.nat_degree ≤ n) :
```

Comment:
Are you sure you want to force `n` to be explicit? Also I think the name should be tweaked:
```suggestion
lemma coeff_add_succ_eq_left_of_le (qn : q.nat_degree ≤ n) :
```

Reply:
I thought that I needed the explicit `n`, but it either was in an earlier version or I misremember!  I am happy to leave it implicit, thanks!